cli/run_util.go

```diff
@@ -1066,3 +1093,39 @@ func runWorkflowWithConfiguration(
 
 	return buildRunResults, nil
 }
+
+func initializeStepDir(additionalEnvironments *[]envmanModels.EnvironmentItemModel) (string, error) {
```

Comment:
Function should not mutate parameter, name could be createTestResultsDir

Reply:
then it wouldn't be a function outsourced but calling an os call in place, I use functions for small logics